### ISSUE TITLE
DummyExtension should use Backend Provider

### ISSUE THREAD

**Issue by Alexander-Krause** [ExplorViz/explorviz-backend#17]
I open this issue here, otherwise we will probably loose track of spread tickets.

Tl;Dr; The source code lines are necessary for extensions to convert Java to JSON-API-conform payload using the backend JSON-API providers.

The [DummyExtension](https://github.com/ExplorViz/explorviz-backend-extension-dummy) uses a seperate provider for JSONAPI. Since the [GenericTypeFinder](https://github.com/ExplorViz/explorviz-backend/blob/master/src/main/java/net/explorviz/server/providers/GenericTypeFinder.java) solves our problem of finding runtime (generic) type information, we can fully use the backend JSONAPI providers.

Use the [DiscoveryExtension](https://github.com/ExplorViz/explorviz-backend-extension-discovery/blob/master/src/main/java/net/explorviz/extension/discovery/main/Application.java) as example for implementing this feature. Additionally add:

```
GenericTypeFinder.typeMap.putIfAbsent("Timestamp", Timestamp.class);
GenericTypeFinder.typeMap.putIfAbsent("Landscape", Landscape.class);
GenericTypeFinder.typeMap.putIfAbsent("System", net.explorviz.model.System.class);
GenericTypeFinder.typeMap.putIfAbsent("NodeGroup", NodeGroup.class);
GenericTypeFinder.typeMap.putIfAbsent("Node", Node.class);
GenericTypeFinder.typeMap.putIfAbsent("Application", Application.class);
GenericTypeFinder.typeMap.putIfAbsent("Component", Component.class);
GenericTypeFinder.typeMap.putIfAbsent("Clazz", Clazz.class);
GenericTypeFinder.typeMap.putIfAbsent("CommunicationClazz", CommunicationClazz.class);
GenericTypeFinder.typeMap.putIfAbsent("Communication", Communication.class);
GenericTypeFinder.typeMap.putIfAbsent("CommunicationAccumulator", CommunicationAccumulator.class);
GenericTypeFinder.typeMap.putIfAbsent("CommunicationTileAccumulator", CommunicationTileAccumulator.class);
GenericTypeFinder.typeMap.putIfAbsent("DatabaseQuery", DatabaseQuery.class);
GenericTypeFinder.typeMap.putIfAbsent("User", User.class);
```

to Extension-Application constructor. GenericTypeFinder is a static class, but it seems that both applications have their own scope. Therefore, they do not share this class.

**Comment by czirkelbach** [comment]
Add API method as a wrapper for enabling core model types in extensions.

**Comment by czirkelbach** [comment]
Testing via an extension necessary

**Comment by Alexander-Krause** [comment]
Works.
